### PR TITLE
$nin filter operator for table scalar dataTypes

### PR DESCRIPTION
**Pull request by Yuqi-Du**
This PR adds $nin support for scalar types, and pair it up with unit test and integration tests.

Also, for TableFilterIntegrationTest.java, I refactor it to parameterized tests and reduce code duplication.

Related issues:
[#1591]
[#1478}

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Review by vkarpov15 — COMMENTED** [review]
LGTM

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/util/DataApiResponseValidator.java [hunk 1]
```diff
@@ -172,6 +172,17 @@ public <T extends APIException> DataApiResponseValidator hasSingleApiError(
     return validator;
   }
 
+  public <T extends APIException> DataApiResponseValidator mayHasSingleApiError(
```
Comment (amorton) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/util/DataApiResponseValidator.java:175]: nit: change mayHasSingleApiError -> mayHaveSingleApiError

Also, if errorCode != null call hasSingleApiError  to remove code duplication

**Review by amorton — APPROVED** [review]
a few nit picking changes, otherise looks good - approved.

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/InTableFilter.java [hunk 2]
```diff
@@ -30,8 +31,8 @@ public class InTableFilter extends TableFilter {
   public final Operator operator;
 
   public enum Operator {
-    IN(BuiltConditionPredicate.IN);
-    // TODO NIN
+    IN(BuiltConditionPredicate.IN),
+    NIN(BuiltConditionPredicate.NIN);
```
[the remark this answers is not in the document]
Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/InTableFilter.java:35]: thanks, deleted!

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/builder/BuiltConditionPredicate.java [hunk 3]
```diff
@@ -8,6 +8,7 @@ public enum BuiltConditionPredicate {
   LTE("<="),
   GTE(">="),
   IN("IN"),
+  NIN("NOT IN"),
```
[the remark this answers is not in the document]
Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/builder/BuiltConditionPredicate.java:11]: deleted.

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/util/DataApiResponseValidator.java (hunk 1 above)
Comment (Yuqi-Du) [reply src/test/java/io/stargate/sgv2/jsonapi/api/v1/util/DataApiResponseValidator.java:175]: fixed

**Pull request merged**